src/graph/rg_matrix/rg_matrix_iter.c

```diff
@@ -92,8 +92,8 @@ GrB_Info RG_MatrixTupleIter_new
 	it->min_row = 0 ;
 	it->max_row = ULLONG_MAX ;
 
-	it->m_it = _init_iter(M, it->min_row, it->max_row, &it->m_depleted) ;
-	it->dp_it = _init_iter(DP, it->min_row, it->max_row, &it->dp_depleted) ;
+	_init_iter(&it->m_it, M, it->min_row, it->max_row, &it->m_depleted) ;
+	_init_iter(&it->dp_it, DP, it->min_row, it->max_row, &it->dp_depleted) ;
```

Comment:
doesn't seems reasonable for the `init` function to accept a pointer to an internal field of the structure it is initializing: `&it->dp_depleted`
consider passing bool depleted?